quickwit/quickwit-storage/src/opendal_storage/google_cloud_storage.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use once_cell::sync::OnceCell;
+use quickwit_common::uri::Uri;
+use quickwit_config::{GoogleCloudStorageConfig, StorageBackend};
+use regex::Regex;
+
+use super::OpendalStorage;
+use crate::debouncer::DebouncedStorage;
+use crate::{Storage, StorageFactory, StorageResolverError};
+
+/// Google cloud storage resolver.
+pub struct GoogleCloudStorageFactory {
+    storage_config: GoogleCloudStorageConfig,
+}
+
+#[async_trait]
+impl StorageFactory for GoogleCloudStorageFactory {
+    fn backend(&self) -> StorageBackend {
+        StorageBackend::Google
+    }
+
+    async fn resolve(&self, uri: &Uri) -> Result<Arc<dyn Storage>, StorageResolverError> {
+        let storage = from_uri(&self.storage_config, uri)?;
```

Comment:
I think to get it to work you also need to add a google protocol in `uri.rs`, and add it in `storage_resolver.rs`

Reply:
Working on it.

Reply:
Addressed.